Add support for multiple rest sensors with a single endpoint

bdraco opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->

Add support for multiple rest sensors with a single endpoint

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/46711
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

bdraco committed:
Add support for multiple rest sensors with a single endpoint

bdraco commented:
I renamed `rest.markdown` to `sensor.rest.markdomain`. It seems to have caused an issue with the rendering but I can't for the life of me figure out why:

```
--- original.markdown	2021-02-17 16:41:52.605941215 -1000
+++ sensor.rest.markdomain	2021-02-17 16:41:57.240974588 -1000
@@ -1,20 +1,17 @@
 ---
-title: RESTful
+title: "RESTful Sensor"
 description: Instructions on how to integrate REST sensors into Home Assistant.
 ha_category:
   - Sensor
 ha_release: 0.7.4
 ha_iot_class: Local Polling
 ha_domain: rest
-ha_platforms:
-  - binary_sensor
-  - notify
-  - sensor
-  - switch
 ---
 
 The `rest` sensor platform is consuming a given endpoint which is exposed by a [RESTful API](https://en.wikipedia.org/wiki/Representational_state_transfer) of a device, an application, or a web service. The sensor has support for GET and POST requests.
 
+_Tip:_ If you want to create multiple `sensors` using the same endpoint, use the [RESTful](/integrations/rest) configuration instructions.
+
 To enable this sensor, add the following lines to your `configuration.yaml` file for a GET request:
 
 ```yaml
```

frenck commented on source/_integrations/rest.markdown:
```diff
@@ -1,10 +1,14 @@
 ---
 title: RESTful
-description: Instructions on how to integrate REST sensors into Home Assistant.
+description: Instructions on how to setup rest sensors within Home Assistant.
```
```suggestion
description: Instructions on how to set up rest sensors within Home Assistant.
```

frenck commented on source/_integrations/rest.markdown:
```diff
@@ -1,10 +1,14 @@
   - Sensor
-ha_release: 0.7.4
+  - Binary Sensor
+ha_release: 2021.3
```
This doesn't change :)
```suggestion
ha_release: 0.7.4
```

frenck commented on source/_integrations/rest.markdown:
```diff
@@ -15,39 +19,98 @@ ha_platforms:
 The `rest` sensor platform is consuming a given endpoint which is exposed by a [RESTful API](https://en.wikipedia.org/wiki/Representational_state_transfer) of a device, an application, or a web service. The sensor has support for GET and POST requests.
 
-To enable this sensor, add the following lines to your `configuration.yaml` file for a GET request:
+[RESTful Sensor](/integrations/sensor.rest) and [RESTful Binary_sensor](/integrations/binary_sensor.rest) can also be setup as platforms if there is only a single sensor per endpoint.
```
```suggestion
[RESTful Sensor](/integrations/sensor.rest) and [RESTful Binary_sensor](/integrations/binary_sensor.rest) can also be set up as platforms if there is only a single sensor per endpoint.
```

frenck commented on source/_integrations/sensor.rest.markdomain:
```diff
@@ -0,0 +1,513 @@
+---
```
We are trying to get away from platform-specific pages. If anything, all other platforms should be merged into `rest.markdown`.

bdraco replied:
That makes sense. If they get to one of the pages they might not discover both options.  Do you want me to merge them in this PR or in a followup?

frenck replied:
I don't know honestly. We don't have to merge all pages right now, but we should not start creating new (old) platform pages.

frenck commented on source/_integrations/rest.markdown:
```diff
@@ -15,39 +19,98 @@ ha_platforms:
-  - platform: rest
-    resource: http://IP_ADDRESS/ENDPOINT
+rest:
+  - authentication: basic
+    username: admin
+    password: password
+    scan_interval: 60
+    resource: http://192.168.1.12/status.xml
```
```suggestion
  - authentication: basic
    username: "admin"
    password: "password"
    scan_interval: 60
    resource: http://192.168.1.12/status.xml
```

frenck commented on source/_integrations/rest.markdown:
```diff
@@ -15,39 +19,98 @@ ha_platforms:
+    scan_interval: 60
+    resource: http://192.168.1.12/status.xml
+    sensor:
+      - name: Adult Pool Data System
```
```suggestion
      - name: "Adult Pool Data System"
```

frenck commented on source/_integrations/rest.markdown:
```diff
@@ -15,39 +19,98 @@ ha_platforms:
+    sensor:
+      - name: Adult Pool Data System
+        json_attributes_path: "$.response.system"
+        value_template: 'OK'
```
```suggestion
        value_template: "OK"
```

frenck commented on source/_integrations/rest.markdown:
```diff
@@ -15,39 +19,98 @@ ha_platforms:
+          - "sensor4"
+          - "sensor5"
+          - "version"
+      - name: Adult Pool Data Equipment
+        json_attributes_path: "$.response.equipment"
+        value_template: 'OK'
```
```suggestion
      - name: "Adult Pool Data Equipment"
        json_attributes_path: "$.response.equipment"
        value_template: "OK"
```

frenck commented on source/_integrations/rest.markdown:
```diff
@@ -15,39 +19,98 @@ ha_platforms:
+          - "circuit6"
+          - "circuit7"
+          - "circuit8"
+      - name: Adult Pool Data Temp
+        json_attributes_path: "$.response.temp"
+        value_template: 'OK'
```
```suggestion
      - name: "Adult Pool Data Temp"
        json_attributes_path: "$.response.temp"
        value_template: "OK"
```

frenck commented on source/_integrations/rest.markdown:
```diff
@@ -15,39 +19,98 @@ ha_platforms:
+          - "spatemp"
+          - "airtemp"
+  - authentication: basic
+    username: admin
+    password: password
+    scan_interval: 60
+    resource: http://192.168.1.13/status.xml
```
```suggestion
    username: "admin"
    password: "password"
    scan_interval: 60
    resource: "http://192.168.1.13/status.xml"
```

frenck commented on source/_integrations/rest.markdown:
```diff
@@ -15,39 +19,98 @@ ha_platforms:
+    scan_interval: 60
+    resource: http://192.168.1.13/status.xml
+    sensor:
+      - name: Kiddie Pool Data System
+        json_attributes_path: "$.response.system"
+        value_template: 'OK'
```
```suggestion
      - name: "Kiddie Pool Data System"
        json_attributes_path: "$.response.system"
        value_template: "OK"
```

frenck commented on source/_integrations/rest.markdown:
```diff
@@ -15,39 +19,98 @@ ha_platforms:
+          - "sensor3"
+          - "sensor4"
+          - "version"
+      - name: Kiddie Pool Data Equipment
+        json_attributes_path: "$.response.equipment"
+        value_template: 'OK'
```
```suggestion
      - name: "Kiddie Pool Data Equipment"
        json_attributes_path: "$.response.equipment"
        value_template: "OK"
```

frenck commented on source/_integrations/rest.markdown:
```diff
@@ -15,39 +19,98 @@ ha_platforms:
+          - "circuit6"
+          - "circuit7"
+          - "circuit8"
+      - name: Kiddie Pool Data Temp
+        json_attributes_path: "$.response.temp"
+        value_template: 'OK'
```
```suggestion
      - name: "Kiddie Pool Data Temp"
        json_attributes_path: "$.response.temp"
        value_template: "OK"
```

bdraco committed:
Update source/_integrations/rest.markdown

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Update source/_integrations/rest.markdown

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Update source/_integrations/rest.markdown

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Update source/_integrations/rest.markdown

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Update source/_integrations/rest.markdown

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Update source/_integrations/rest.markdown

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Update source/_integrations/rest.markdown

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Update source/_integrations/rest.markdown

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Update source/_integrations/rest.markdown

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Update source/_integrations/rest.markdown

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Update source/_integrations/rest.markdown

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Update source/_integrations/rest.markdown

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco commented:
hmm

The deploy still looks mangled

https://deploy-preview-16632--home-assistant-docs.netlify.app/integrations/sensor.rest/

I'm at a loss as to whats amiss.  I'll have to come back to this as my work day is getting busy

bdraco committed:
Merge branch 'next' into rest_single_endpoint

bdraco commented on source/_integrations/sensor.rest.markdomain:
```diff
@@ -0,0 +1,513 @@
+            data:
+               led: 6
+          - service: homeassistant.update_entity
+            data:
```
```suggestion
            target:
```

bdraco commented on source/_integrations/sensor.rest.markdomain:
```diff
@@ -0,0 +1,513 @@
+               entity_id: sensor.steam_system_data
+          - delay: 00:00:15
+          - service: homeassistant.update_entity
+            data:
```
```suggestion
            target:
```

bdraco commented on source/_integrations/sensor.rest.markdomain:
```diff
@@ -0,0 +1,513 @@
+            data:
+               led: 7
+          - service: homeassistant.update_entity
+            data:
```
```suggestion
            target:
```

bdraco commented on source/_integrations/sensor.rest.markdomain:
```diff
@@ -0,0 +1,513 @@
+               entity_id: sensor.steam_system_data
+          - delay: 00:00:15
+          - service: homeassistant.update_entity
+            data:
```
```suggestion
            target:
```

bdraco committed:
Update source/_integrations/sensor.rest.markdomain

bdraco committed:
Update source/_integrations/sensor.rest.markdomain

bdraco committed:
Update source/_integrations/sensor.rest.markdomain

bdraco committed:
Update source/_integrations/sensor.rest.markdomain

bdraco committed:
fix name conflict

bdraco committed:
ok... I think I am losing it markdomain -> markdown

bdraco commented:
<img width="665" alt="Screen Shot 2021-02-23 at 9 58 53 PM" src="https://user-images.githubusercontent.com/663432/108945522-5b708b00-7622-11eb-83f8-3d5f371fa50f.png">

After I fixed my braindead file naming mistake, I fixed the name conflict so at least we can tell them apart now.

bdraco commented on source/_integrations/rest.markdown:
```diff
@@ -112,405 +158,13 @@ headers:
-json_attributes_path:
-  description: A [JSONPath](https://goessner.net/articles/JsonPath/) that references the location of the `json_attributes` in the JSON content.
+sensor:
+  description: A list of [RESTful Sensor](/integrations/sensor.rest) to create from the shared data. All configuration settings that the supported by `RESTful Sensor` not listed above can be used here.
```
```suggestion
  description: A list of [RESTful Sensor](/integrations/sensor.rest) to create from the shared data. All configuration settings that the supported by [RESTful Sensor](/integrations/sensor.rest#configuration-variables) not listed above can be used here.
```

bdraco commented on source/_integrations/rest.markdown:
```diff
@@ -112,405 +158,13 @@ headers:
-  description: Sends update events even if the value hasn't changed. Useful if you want to have meaningful value graphs in history.
+  type: list
+binary_sensor:
+  description: A list of [RESTful Binary_sensor](/integrations/binary_sensor.rest) to create from the shared data All configuration settings that the supported by `RESTful Binary Sensor` not listed above can be used here.
```
```suggestion
  description: A list of [RESTful Binary_sensor](/integrations/binary_sensor.rest#configuration-variables) to create from the shared data All configuration settings that the supported by `RESTful Binary Sensor` not listed above can be used here.
```

bdraco replied:
```suggestion
  description: A list of [RESTful Binary_sensor](/integrations/binary_sensor.rest) to create from the shared data All configuration settings that the supported by [RESTful Binary_sensor](/integrations/binary_sensor.rest#configuration-variables) not listed above can be used here.
```

bdraco committed:
Update source/_integrations/rest.markdown

bdraco committed:
Update source/_integrations/rest.markdown

frenck commented:
> After I fixed my braindead file naming mistake, I fixed the name conflict so at least we can tell them apart now.

That won't work, as the titles are synced from the codebase (originate from integration manifests).
I suggest merging this, and I will look into merging the pages before the end of beta.
